app/controllers/articles_controller.rb

```diff
@@ -34,58 +34,35 @@ def feed
 
   def new
     base_editor_assigments
-    @article = if @tag.present? && @user&.editor_version == "v2"
-                 authorize Article
-                 submission_template = @tag.submission_template_customized(@user.name).to_s
-                 Article.new(
-                   body_markdown: submission_template.split("---").last.to_s.strip,
-                   cached_tag_list: @tag.name,
-                   processed_html: "",
-                   user_id: current_user&.id,
-                   title: submission_template.split("title:")[1].to_s.split("\n")[0].to_s.strip,
-                 )
-               elsif @tag&.submission_template.present? && @user
-                 authorize Article
-                 Article.new(
-                   body_markdown: @tag.submission_template_customized(@user.name),
-                   processed_html: "",
-                   user_id: current_user&.id,
-                 )
-               elsif @prefill.present? && @user&.editor_version == "v2"
-                 authorize Article
-                 Article.new(
-                   body_markdown: @prefill.split("---").last.to_s.strip,
-                   cached_tag_list: @prefill.split("tags:")[1].to_s.split("\n")[0].to_s.strip,
-                   processed_html: "",
-                   user_id: current_user&.id,
-                   title: @prefill.split("title:")[1].to_s.split("\n")[0].to_s.strip,
-                 )
-               elsif @prefill.present? && @user
-                 authorize Article
-                 Article.new(
-                   body_markdown: @prefill,
-                   processed_html: "",
-                   user_id: current_user&.id,
-                 )
-               elsif @tag.present?
-                 skip_authorization
-                 Article.new(
-                   body_markdown: "---\ntitle: \npublished: false\ndescription: \ntags: #{@tag.name}\n---\n\n",
-                   processed_html: "",
-                   user_id: current_user&.id,
-                 )
-               else
-                 skip_authorization
-                 if @user&.editor_version == "v2"
-                   Article.new(user_id: current_user&.id)
-                 else
-                   Article.new(
-                     body_markdown: "---\ntitle: \npublished: false\ndescription: \ntags: \n---\n\n",
-                     processed_html: "",
-                     user_id: current_user&.id,
-                   )
-                 end
-               end
+    article_builder = Articles::Builder.new(current_user&.id, @user&.name, @tag, @prefill)
+
+    if @tag.present? && @user&.editor_version == "v2"
```

Comment:
I wonder if this should move into the builder too? To me it'd make more sense to call something like 

```ruby
@article = Articles::Builder.new(current_user&.id, @user&.name, @tag, @prefill).build
```

or something and be done with it, rather than having this functionality split between builder and controller. 

You can manually call the policy with something like

```ruby
ArticlePolicy.new(current_user, @article).new?
```

inside the builder or return two values, e.g. 

```ruby
@article, needs_authorization = Articles::Builder.new(current_user&.id, @user&.name, @tag, @prefill).build
```

whichever you like better.

Reply:
Good call. I'll do that.